### PR TITLE
Fix Information Exposure Through Error Message Vulnerability

### PR DESCRIPTION
**Pull request by dealako**
* Reviewed flagged log messages, updated format and content of messages, and annotated the Checkmarx tool. Log messages were in one of three categories:
* Log information is saved to the private system log and visible in the job log. Information is critical to diagnosing and reporting error conditions with the Jenkins plugin and not deemed exploitable.
* Log information is saved to the private system log and not visible to public viewers. Information is critical to diagnosing error conditions with the Jenkins plugin.
* Log information is saved to the private system log and visible in the test log. Information is critical to diagnosing and reporting error conditions with the Jenkins plugin and not deemed exploitable.

JIRA: RE-2197

### CODE REVIEW

File: src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java [hunk 1]
```diff
@@ -100,42 +128,42 @@ public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskLi
...
-            log("Can't hold as theres no node allocated :(");
+            log("Can't hold as there are no nodes allocated :(");
             return;
         }
 
-        if (computer instanceof NodePoolComputer){
+        if (computer instanceof NodePoolComputer) {
+            NodePoolComputer npc = (NodePoolComputer) computer;
+            NodePoolSlave nps = (NodePoolSlave) npc.getNode();
             try {
```
Comment (hughsaunders) [inline src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java:145]: Why move the try?

**Review by hughsaunders — COMMENTED** [review]
Is the idea that you shouldn't pass an exception object directly to a log call, but passing its properties through format is ok? 

Added inline comment about moving a try.

**Review by hughsaunders — APPROVED** [review]

File: src/main/java/com/rackspace/jenkins_nodepool/Janitor.java [hunk 2]
```diff
@@ -92,7 +92,7 @@ private void runAsSystem() {
                 Thread.sleep(sleepMilliseconds);
                 clean();
             } catch (Exception e) {
-                LOG.log(WARNING, "Cleanup failed: " + e.getMessage(), e);
+                LOG.log(WARNING, format("%s while running cleanup. Message: ", e.getClass().getSimpleName(), e.getLocalizedMessage()));
```
Comment (hughsaunders) [inline src/main/java/com/rackspace/jenkins_nodepool/Janitor.java:95]:
```suggestion
                LOG.log(WARNING, format("%s while running cleanup. Message: %s", e.getClass().getSimpleName(), e.getLocalizedMessage()));
```

**Review by hughsaunders — CHANGES_REQUESTED** [review]

File: src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java (hunk 1 above)
Comment (dealako) [reply src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java:145]: So I could get access to the NPC object to print out the node name if an exception occurs.  Just printing a stack trace will work, but a meaningful error message with details is better.

**Pull request closed without merging**